extension/src/lib/portBinding.js

```diff
@@ -69,14 +288,30 @@
                 type: "fluid.dataSource.encoding.model"
             }
         },
+        members: {
+            lastIncomingPayload: {}
```

Comment:
I think that this particular element of the dataStore isn't useful for general clients and should be moved into the testing grade

Reply:
It's actually used as part of the get mechanism in the "onRead.impl" to return the value. Essentially this is an in memory store of the latest updates sent via the messaging system.

Reply:
Seems suspicious - shouldn't it be that the store actually issues reads in order to honour requests? In which case the read implementation should just consist of a promise which resolves at the point the appropriate payload response is received. If people want to perform caching based on such a DataSource that should be implemented at the DataSource level rather than intruding on the messaging level. Do we have a requirement for non-live reads?

Reply:
Spoke to @amb26 in the fluid-work channel about this.

In summary it is suggested to remove the "store" and use the DataSource API with the gpii.chrome.portBinding to directly get/set model values.

see: https://botbot.me/freenode/fluid-work/2018-09-26/?msg=104841751&page=2